lib/broadway/producer.ex

```diff
@@ -62,7 +62,7 @@ defmodule Broadway.Producer do
 
   @impl true
   def handle_call({:push_messages, messages}, _from, state) do
-    {:reply, :ok, messages, state}
+    {:reply, :ok, transform_events(messages, state.transformer), state}
```

Comment:
Actually, I wouldn't the transformer here. Imagine if you are expecting to receive numbers as events. Now when testing you also need to consider that it may receive Broadway.Message. Maybe a better way to test this is to call `GenStage.stop` or `GenServer.stop` in the producer?

Reply:
Yes, we can do that. Also, I still think we need to test what happens when the transformer fails. So I'll create another set of tests for it. Probably need to create another producer so we make sure the tests hit this lines: https://github.com/plataformatec/broadway/blob/8ee0bdd922bd578360a99d044ced58825437d6a4/lib/broadway/producer.ex#L29
https://github.com/plataformatec/broadway/blob/8ee0bdd922bd578360a99d044ced58825437d6a4/lib/broadway/producer.ex#L33

Same for `handle_info`.